gunicorn/_compat.py

```diff
@@ -62,147 +58,20 @@ def _get_codeobj(pyfile):
-        cached = _parse_cache.get(key, None)
-        if cached:
-            return cached
-        if len(_parse_cache) >= MAX_CACHE_SIZE: # avoid runaway growth
-            clear_cache()
-        netloc = query = fragment = ''
-        i = url.find(':')
-        if i > 0:
-            if url[:i] == 'http': # optimize the common case
-                scheme = url[:i].lower()
-                url = url[i+1:]
-                if url[:2] == '//':
-                    netloc, url = _splitnetloc(url, 2)
-                    if (('[' in netloc and ']' not in netloc) or
-                            (']' in netloc and '[' not in netloc)):
-                        raise ValueError("Invalid IPv6 URL")
-                if allow_fragments and '#' in url:
-                    url, fragment = url.split('#', 1)
-                if '?' in url:
-                    url, query = url.split('?', 1)
-                v = SplitResult(scheme, netloc, url, query, fragment)
-                _parse_cache[key] = v
-                return v
-            for c in url[:i]:
-                if c not in scheme_chars:
-                    break
-            else:
-                # make sure "url" is not actually a port number (in which case
-                # "scheme" is really part of the path)
-                rest = url[i+1:]
-                if not rest or any(c not in '0123456789' for c in rest):
-                    # not a port number
-                    scheme, url = url[:i].lower(), rest
-
-        if url[:2] == '//':
-            netloc, url = _splitnetloc(url, 2)
-            if (('[' in netloc and ']' not in netloc) or
-                    (']' in netloc and '[' not in netloc)):
-                raise ValueError("Invalid IPv6 URL")
-        if allow_fragments and '#' in url:
-            url, fragment = url.split('#', 1)
-        if '?' in url:
-            url, query = url.split('?', 1)
-        v = SplitResult(scheme, netloc, url, query, fragment)
-        _parse_cache[key] = v
-        return v
-
-else:
-    from gunicorn.six.moves.urllib.parse import urlsplit
-
-
-import inspect
-
-if hasattr(inspect, 'signature'):
-    positionals = (
-        inspect.Parameter.POSITIONAL_ONLY,
-        inspect.Parameter.POSITIONAL_OR_KEYWORD,
-    )
-
-    def get_arity(f):
-        sig = inspect.signature(f)
-        arity = 0
-
-        for param in sig.parameters.values():
-            if param.kind in positionals:
-                arity += 1
-
-        return arity
-else:
-    def get_arity(f):
-        return len(inspect.getargspec(f)[0])
-
-
-try:
-    import html
-
-    def html_escape(s):
-        return html.escape(s)
-except ImportError:
-    import cgi
-
-    def html_escape(s):
-        return cgi.escape(s, quote=True)
+
+def execfile_(fname, *args):
+    if fname.endswith(".pyc"):
+        code = _get_codeobj(fname)
+    else:
+        code = compile(open(fname, 'rb').read(), fname, 'exec')
+    return exec(code, *args)
+
+
+def bytes_to_str(b):
```

Comment:
We can now move `bytes_to_str()` and `unquote_to_wsgi_str()` to `gunicorn.util` since they no longer behave differently in Python 2 and Python 3.

`execfile_()` and its friends can stay here for now. There was a discussion about deprecating PYC support in 20 and rewriting it with importlib (instead of deprecated) is a complex task.